### PR TITLE
Allow user to add httpOptions in the config

### PR DESCRIPTION
**Pull request by CaptainGlac1er**
## About the changes

Allows adding in httpOptions which then gets passed to the unleash client.
This enables
- Using self-signed certs (successfully tested with GitLab) 

It should be setup that if the user doesn't add a custom httpOptions it would maintain the same runtime before this merge request. 

Let me know if changes need to be made! I tried to minimize the changes best I could. :) 

Closes # .

### Important files

src/client.ts
src/config.ts

## Discussion points

Just want to make sure this isn't a breaking change or if that is the right place to add in the config object.

### CODE REVIEW

File: src/test/config.test.ts [hunk 1]
```diff
@@ -339,14 +339,25 @@ test('should load cors origin, maxAge and exposedHeaders default values', () =>
...
     });
-    expect(config.httpOptions?.agent?.(new URL('https://example.com'))).toBe(https.globalAgent);
-})
\ No newline at end of file
+    expect(config.httpOptions?.agent?.(new URL('https://example.com'))).toBe(
+        https.globalAgent,
+    );
+});
+
+test('when not passed with httpOptions the config should not contain that property', () => {
```
Comment (thomasheartman) [inline src/test/config.test.ts:356]: Minor test description rewording suggestion 😄 

```suggestion
test('should not set config.httpOptions if no http options are provided at creation', () => {
```

File: src/test/config.test.ts [hunk 2]
```diff
@@ -339,14 +339,25 @@ test('should load cors origin, maxAge and exposedHeaders default values', () =>
     expect(config.cors.exposedHeaders).toBe('ETag');
 });
 
-test('should load passed in http agent', () => {
+test('when passed with agent in httpOptions config.httpOptions.agent should be callable with url', () => {
```
Comment (thomasheartman) [inline src/test/config.test.ts:342]:
```suggestion
test('should load the passed-in http agent when config.httpOptions is provided', () => {
```

**Review by thomasheartman — CHANGES_REQUESTED** [review]
Sounds like a good use case and the changes look sensible to me 🙌🏼  Thank you so much for putting in the effort! I've added some minor change suggestions for the test descriptions, but that's it from my side. However, I'll need an extra set of eyes to confirm that this is all good otherwise.

Thanks again!

**Review by thomasheartman — APPROVED** [review]
Nice one! Thanks for the quick fixup 😄 That's all from my end. I'm still waiting for someone more familiar with the proxy to give it a go-ahead, but should be fine, I reckon 🥳

**Comment by CaptainGlac1er** [comment]
@thomasheartman sweet thanks!

**Review by sighphyre — APPROVED** [review]
Thanks! This looks good to me

**Pull request merged**